### PR TITLE
✨ Load Order

### PR DESCRIPTION
**Pull request by effs**
Related: #231 details some complexity around enabled status

### CODE REVIEW

File: src/load_order.ts [hunk 1]
```diff
@@ -22,14 +56,39 @@ import {
...
 const path = win32;
 
+
+// Defs
+
+const LOAD_ORDER_FILE_NAME =
+  `${EXTENSION_NAME_INTERNAL}-redmod-load-order.json`;
+
+const LOAD_ORDER_FILE_RELATIVE_PATH =
+  path.join(`r6\\cache\\modded\\${LOAD_ORDER_FILE_NAME}`);
```
Comment (E1337Kat) [inline src/load_order.ts:77]: I don't think we need this. This seems to me like we are gonna use our own file, but I think it is better to use the file that already exists from the redmod. This simplifies things and won't cause issues with the UI and file being incongruous.

Comment (E1337Kat) [reply src/load_order.ts:77]: I mean we should be using `const modListPath = path.join(discoveryPath, 'r6', 'cache', 'modded', 'mods.json');` for the load order file, and deserialize and serialize to it.

File: src/load_order.ts [hunk 2]
```diff
@@ -211,35 +349,60 @@ export const internalDeserializeLoadOrder: VortexWrappedDeserializeFunc = async
...
+      modId: entry.modId,
+      displayName: entry.name,
+      enabledInVortex: entry.enabled,
+      dirPath: entry.data?.dirPath || `unknown path`,
+      version: entry.data?.version || `unknown version`,
+    })),
+    toMutableArray,
+  );
+
+  const loadOrder: LoadOrder = {
```
Comment (E1337Kat) [inline src/load_order.ts:381]: we don't need to keep any special meta infomation in the file. it should _just_ be a list of mods in order from 0-n where mods[0] is loaded first and mods[n] is loaded last.

File: src/load_order.ts [hunk 3]
```diff
@@ -211,35 +349,60 @@ export const internalDeserializeLoadOrder: VortexWrappedDeserializeFunc = async
...
+    typeVersion: LOAD_ORDER_TYPE_VERSION,
+    generatedAt: new Date().toISOString(),
+    entriesInOrderWithEarlierWinning: loadOrderEntries,
+  };
+
+  const serializedLoadOrder =
+    encodeLoadOrder(loadOrder);
+
+  const loadOrderFilePath =
+    path.join(discoveryPath, LOAD_ORDER_FILE_RELATIVE_PATH);
```
Comment (E1337Kat) [inline src/load_order.ts:391]: use the mods.json file

File: src/load_order.ts (hunk 1 above)
Comment (effs) [reply src/load_order.ts:77]: We have to run `redmod deploy`, which generates that file. It is needed for script compilation plus it’s possible it does validation that we don’t want to reproduce.

File: src/load_order.ts (hunk 2 above)
Comment (effs) [reply src/load_order.ts:381]: You always need metadata :) If not now, then when we realize the format needs to change.

**Comment by E1337Kat** [comment]
You have gotten to the point that it is hard for me to follow the logic lol. Good stuff though. When it gets usable, let me know. I want to add in an options panel to enable disable the various features.

File: src/index.ts [hunk 4]
```diff
@@ -82,6 +140,74 @@ const main = (vortex: VortexExtensionContext) => {
...
+        [br][/br]
+        The load order is saved automatically, and will be deployed whenever the next
+        Vortex deployment occurs - you can also manually click to deploy, if you like!
+        [br][/br]
+        REDmod deployment can take a little while if you have tweak or script mods,
+        so wait for the green success notification before you start the game! :)
+        [br][/br]
+        You can still use the redmod tool manually, too, but changes won't be reflected
+        in Vortex.
+      `),
```
Comment (E1337Kat) [inline src/index.ts:180]: I like this. It's good to have a nice explanation. Hopefully it mitigates the issues that I see happening... assuming people read.

File: src/installers.utils.ts [hunk 5]
```diff
@@ -17,3 +20,20 @@ export const jsonpp = (thing: unknown): string => JSON.stringify(thing, null, 2)
...
+);
+
+
+<<<<<<< HEAD
+export const bbcodeBasics = flow(
+  replaceIn(/\n{2,}/g, `\n[br][/br][br][/br]\n`), // Any number of empty lines becomes a single bbcode line break
+=======
+export const bbcodeParagraphs = flow(
+  replaceIn(/\n{2,}/g, `[br][/br][br][/br]`), // Any number of empty lines becomes a single bbcode line break
+>>>>>>> 833e033 (🔧📝🏷️ Fix schema dump)
```
Comment (E1337Kat) [inline src/installers.utils.ts:38]:
```suggestion
export const bbcodeBasics = flow(
  replaceIn(/\n{2,}/g, `\n[br][/br][br][/br]\n`), // Any number of empty lines becomes a single bbcode line break
  
export const bbcodeParagraphs = flow(
  replaceIn(/\n{2,}/g, `[br][/br][br][/br]`), // Any number of empty lines becomes a single bbcode line break
```

Merge conflict!!

Comment (effs) [reply src/installers.utils.ts:38]: I was obviously very tired XD

File: src/load_order.ts [hunk 6]
```diff
@@ -0,0 +1,648 @@
...
+// Previous load order is used to establish the earlier order, and
+// any new ones are added to the end. This could also be done the other
+// way around, but didn't.
+//
+// Unmanaged mods are currently NOT SUPPORTED, maybe TODO:
+// https://github.com/E1337Kat/cyberpunk2077_ext_redux/issues/264
+//
+
+
+const compileDetesToGenerateLoadOrderUi: VortexWrappedDeserializeFunc = async (
```
[the remark this answers is not in the document]
Comment (effs) [reply src/load_order.ts:288]: Yep, you’re absolutely right, it doesn’t quite reflect the exact purpose.

File: src/installers.utils.ts (hunk 5 above)
Comment (E1337Kat) [reply src/installers.utils.ts:38]: lol, it happens ;)

**Pull request merged**